base/shell/explorer/traywnd.cpp

```diff
@@ -365,7 +366,15 @@ class CTrayWindow :
 
         m_RunFileDlgOwner = hwnd;
 
-        RunFileDlg(hwnd, NULL, NULL, NULL, NULL, RFF_CALCDIRECTORY);
+        WCHAR szHomeDrive[MAX_PATH], szHomePath[MAX_PATH];
+        GetEnvironmentVariableW(L"HOMEDRIVE", szHomeDrive, _countof(szHomeDrive));
+        GetEnvironmentVariableW(L"HOMEPATH", szHomePath, _countof(szHomePath));
```

Comment:
Why not SHGetFolderPath(... CSIDL_PROFILE...)?

Reply:
According to API Monitor investigation, 2k3 is doing so.